projects/plugins/jetpack/tools/check-block-assets.php

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env php
+<?php
+/**
+ * Script to check blocks' view.asset.php for unexpected dependencies.
+ *
+ * @package automattic/jetpack
+ */
+
+// phpcs:disable WordPress.Security.EscapeOutput.OutputNotEscaped -- This is not WordPress code.
+
+/**
+ * Non-WP dependencies to check for.
+ *
+ * @var string[]
+ */
+$bad_deps = array(
+	'jquery',
+	'lodash',
+	'lodash-es',
+	'moment',
+	'react',
+	'react-dom',
+);
+
+/**
+ * WP dependencies to allow.
+ *
+ * @var string[]
+ */
+$ok_wp_deps = array(
```

Comment:
I went with a whitelist here versus the above blacklist since it seems likely new packages should be checked before being allowed.

This list consists of all the Gutenberg packages that are registered with WordPress and that don't depend on anything in `$bad_deps`. Plus `wp-polyfill`, which is often added by the extraction plugin.